jre.go

```diff
@@ -149,3 +162,34 @@ func (j JRE) Contribute(layer libcnb.Layer) (libcnb.Layer, error) {
 func (j JRE) Name() string {
 	return j.LayerContributor.LayerName()
 }
+
+func (j JRE) buildCustomJRE(layerPath string) error {
+	buildpackPath := filepath.Dir(layerPath)
+	customDir := filepath.Join(buildpackPath, "custom-jre")
+
+	def := fmt.Sprintf("--add-modules ALL-MODULE-PATH --no-man-pages --no-header-files --strip-debug --output %s", customDir)
+	jlinkArgs := sherpa.GetEnvWithDefault("BP_JVM_JLINK_ARGS", def)
+	if jlinkArgs != def {
+		if strings.Contains(jlinkArgs, "output") {
+			return fmt.Errorf("invalid custom jlink config, please remove 'output' arg\n")
+		}
+		jlinkArgs = fmt.Sprintf("%s --output %s", jlinkArgs, customDir)
+	}
+
+	if err := j.Executor.Execute(effect.Execution{
+		Command: filepath.Join(layerPath, "bin", "jlink"),
+		Args:    strings.Split(jlinkArgs, " "),
+	}); err != nil {
+		return fmt.Errorf("error running jlink\n%w", err)
+	}
+	if err := os.Rename(layerPath, filepath.Join(buildpackPath, "original-jdk")); err != nil {
```

Comment:
This rename, rename, delete seems a little odd.

I wonder if it might make sense to split this into two different LayerContributors. We could then skip installing the JRE altogether if the JDK has already been installed. Run `jlink` to produce what would otherwise be the JRE layer. 

If it's not a JDK, then we just install the JRE like normal (I didn't look, but I'm assuming `jlink` is only part of the JDK so we couldn't use it without a JDK being present.

That brings up an edge case. What if the user sets the JVM type to JDK. Normally, that would force the JDK at both build and runtime. If they also enable `jlink` would that then install the `jlink`ed version at runtime? As I'm thinking about it, I feel like that would be the right answer. I can't think of why you'd force a JDK, enable jlink and not expect that, but it's late here so i could be missing something :)